src/Microsoft.AspNet.Mvc.Core/Infrastructure/ObjectResultExecutor.cs

```diff
@@ -314,22 +319,20 @@ public virtual Task ExecuteAsync(ActionContext context, ObjectResult result)
             {
                 throw new ArgumentNullException(nameof(sortedAcceptHeaders));
             }
-
-            IOutputFormatter selectedFormatter = null;
+            
             foreach (var contentType in sortedAcceptHeaders)
             {
-                // Loop through each of the formatters and see if any one will support this
-                // mediaType Value.
-                selectedFormatter = formatters.FirstOrDefault(
-                    formatter => formatter.CanWriteResult(formatterContext, contentType));
-                if (selectedFormatter != null)
+                foreach (var formatter in formatters)
                 {
-                    // we found our match.
-                    break;
+                    formatterContext.ContentType = contentType;
```

Comment:
Suspect `Debug.Assert(formatterContext.ContentType == contentType); at the bottom of this loop would match the intent better. Then can init `ContentType` once.